services/accumulo-service/src/main/java/datawave/microservice/accumulo/lookup/config/LookupAuditProperties.java

```diff
@@ -0,0 +1,179 @@
+            
+            this.tableRegex = tableRegex;
+            if (rowRegex != null) {
+                this.rowRegex = rowRegex;
+            }
+            if (colFamRegex != null) {
+                this.colFamRegex = colFamRegex;
+            }
+            if (colQualRegex != null) {
+                this.colQualRegex = colQualRegex;
+            }
+            this.auditType = auditType;
+            
+            validate();
+        }
+        
+        public void validate() throws IllegalArgumentException {
+            
+            if (this.tableRegex == null) {
+                throw new IllegalArgumentException("tableRegex can not be null");
+            }
+            if (this.auditType == null) {
+                throw new IllegalArgumentException("auditType can not be null");
+            }
+        }
+        
+        public String getTableRegex() {
+            return tableRegex;
+        }
+        
+        public void setTableRegex(String tableRegex) {
+            this.tableRegex = tableRegex;
+        }
+        
+        public String getRowRegex() {
+            return rowRegex;
+        }
+        
+        public void setRowRegex(String rowRegex) {
+            this.rowRegex = rowRegex;
+        }
+        
+        public String getColFamRegex() {
+            return colFamRegex;
+        }
+        
+        public void setColFamRegex(String colFamRegex) {
+            this.colFamRegex = colFamRegex;
+        }
+        
+        public String getColQualRegex() {
+            return colQualRegex;
+        }
+        
+        public void setColQualRegex(String colQualRegex) {
+            this.colQualRegex = colQualRegex;
+        }
+        
+        public AuditType getAuditType() {
+            return auditType;
+        }
+        
+        public void setAuditType(AuditType auditType) {
+            this.auditType = auditType;
+        }
+        
+        public boolean isMatch(String table, String row, String colFam, String colQual) {
+            
+            boolean isMatch = false;
+            
+            // table == null should never happen, but this protects against a NPE
+            if (table == null) {
+                table = "";
+            }
+            // if row == null, we still want it to match against a rowRegex of .*
+            if (row == null) {
+                row = "";
+            }
+            // if colFam == null, we still want it to match against a colFamRegex of .*
+            if (colFam == null) {
+                colFam = "";
+            }
+            // if colQual == null, we still want it to match against a colQualRegex of .*
+            if (colQual == null) {
+                colQual = "";
+            }
+            
+            if (this.tableRegex != null && this.rowRegex != null && this.colFamRegex != null && this.colQualRegex != null) {
```

Comment:
You may need to check for empty values as well. You have a default of ".*" set for a number of these, but then a value could be set via property binding. I don't know of a way to set null (since it will turn into the string "null"), which means the empty string is the only way to override.
Not a likely case, I know, but I'd say either handle it or change the validation to not allow null/blank.

Reply:
Fixed in latest commit